File: src/main/java/org/cerion/stocklist/data/MACDArray.java
package org.cerion.stocklist.data;

public class MACDArray extends FloatArray {

	private FloatArray mSignal;
	
	public MACDArray(int size) { super(size); }

	//TODO, see if this can be done in constructor
	public void setSignal(int period)
	{
		mSignal = ema(period);
	}
	
	public float signal(int pos) //Signal line
	{
		return mSignal.get(pos);
	}
	
	public float hist(int pos) //Histogram
	{
		return mVal[pos] - mSignal.get(pos);
	}
}

File: src/main/java/org/cerion/stocklist/arrays/MACDArray.kt
package org.cerion.stocklist.arrays

class MACDArray(size: Int, private val signalPeriod: Int) : FloatArray(size) {

    private val signal: FloatArray by lazy {
        ema(signalPeriod)
    }

    //Signal line
    fun signal(pos: Int) : Float = signal.get(pos)

    //Histogram
    fun hist(pos: Int) : Float = this[pos] - signal.get(pos)
}
